src/ControlSystem/ControlErrors/SizeControlState.hpp

```diff
@@ -0,0 +1,95 @@
+  // is in state SizeControlLabel::DeltaR.
+  // This is Q in Eq. 96 of ArXiv:1211.6079.
+  double control_error_delta_r;
+};
+
+/// Packages some of the inputs to the SizeControlState::control_signal, so
+/// that SizeControlState::control_signal doesn't need a large number of
+/// arguments.
+struct SizeControlStateControlSignalArgs {
+  SizeControlStateControlSignalArgs(const double min_speed,
+                                    const double control_err_delta_r,
+                                    const double avg_normal_dot_unit_vector,
+                                    const double dt_lambda_00)
+      : min_char_speed(min_speed),
+        control_error_delta_r(control_err_delta_r),
+        avg_distorted_normal_dot_unit_coord_vector(avg_normal_dot_unit_vector),
+        time_deriv_of_lambda_00(dt_lambda_00) {}
+  double min_char_speed;
+  double control_error_delta_r;
+  // avg_distorted_normal_dot_unit_coord_vector is the average of
+  // distorted_normal_dot_unit_coord_vector over the excision
+  // boundary.  Here distorted_normal_dot_unit_coord_vector is Eq. 93
+  // of ArXiv:1211.6079.  distorted_normal_dot_unit_coord_vector is
+  // nhat_i x^i/r where nhat_i is the distorted-frame unit normal to
+  // the excision boundary (pointing INTO the hole, i.e. out of the
+  // domain), and x^i/r is the distorted-frame (or equivalently the
+  // grid frame because it is invariant between these two frames
+  // because of the required limiting behavior of the map we choose)
+  // Euclidean normal vector from the center of the excision-boundary
+  // Strahlkorper to each point on the excision-boundary Strahlkorper.
+  double avg_distorted_normal_dot_unit_coord_vector;
+  // time_deriv_of_lambda_00 is the time derivative of the quantity lambda_00
+  // that appears in ArXiv:1211.6079.  time_deriv_of_lambda_00 is (minus)
+  // the radial velocity of the excision boundary in the distorted frame with
+  // respect to the grid frame.
+  double time_deriv_of_lambda_00;
+};
+
+/// Represents a 'state' of the size control system.
+///
+/// Each 'state' of the size control system has a different control
+/// signal, which has a different purpose, even though each state
+/// controls the same map quantity, namely the Y00 coefficient of the
+/// shape map.  For example, state SizeControlLabel::AhSpeed controls
+/// the Y00 coefficient of the shape map so that the minimum
+/// characteristic speed is driven towards a target value, and state
+/// SizeControlLabel::DeltaR controls the Y00 coefficient of the shape
+/// map so that the minimum difference between the horizon radius and
+/// the excision boundary radius is driven towards a constant.
+///
+/// Each state has its own logic (the 'update' function) that
+/// determines values of certain parameters (i.e. the things in
+/// SizeControlInfo), including whether the control system should
+/// transition to a different state.
+class SizeControlState {
```

Comment:
All of these classes you've added are in the global namespace. Again, if you don't have too much future code that depends on this PR, could you move all these classes into a new namespace `SizeControl` or `Size` that's within the `control_system::ControlErrors` namespace? (also removing the "SizeControl" prefix from all the classes)